web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -59,20 +58,24 @@
     private boolean agentIsUp = false;
     private final LinkedList<ThreadLocalCleaner> cleaners = new LinkedList<ThreadLocalCleaner>();
 
+    //Only for Spring Boot Application (External Name Setting)
+    private String applicationName;
```

Comment:
@SergeyKanzhelev I just looked into the code and it seems that we cannot make the applicationName final because if we do so it has to be initialized in the constructor itself (either through parameterized constructor or default constructor) and we do not have a way to do it with default constructor because with non spring boot application the name of the application is caught from the servlet context and in our implementation it is retrieved from getName() method which parses the context and is invoked post constructor call. Let me know.